### ISSUE TITLE
[Feature Enhancement]: Add Accessibility to the SBBStepper

### ISSUE THREAD

**Issue by smallTrogdor** [SchweizerischeBundesbahnen/design_system_flutter#713]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/SchweizerischeBundesbahnen/design_system_flutter/blob/main/Contributing.md) for this project.
- [x] I have searched the [issue tracker](https://github.com/orgs/...) for an issue that matches the one I want to file, without success.

### Request type

Request for enhancement of a component

### Functionality

As of v5 of the DSM, the SBBStepper does not support screen readers properly.

The screen reader should announce the selected index correctly and allow for semantic labels of the individual steps.
